source/simulator/assemblers/stokes.cc

```diff
@@ -56,6 +56,52 @@ namespace aspect
           ++i;
         }
 
+      // When using the Q1-Q1 equal order element, we need to compute the
+      // projection of the Q1 pressure shape functions onto the constants
+      // and use this projection in the computation of matrix terms.
+      // Do this here by computing the integral of the shape functions
+      // over the cell and then dividing by the area of the cell.
+      std::vector<double> average_pressure_shape_function (this->get_parameters().use_equal_order_interpolation_for_stokes
+                                                           ?
+                                                           stokes_dofs_per_cell
+                                                           :
+                                                           0,
+                                                           numbers::signaling_nan<double>());
+      if (this->get_parameters().use_equal_order_interpolation_for_stokes)
+        {
+          // Check that we are really only using a Q1-Q1 element and
+          // not a Q2-Q2 element. This is because in the latter case, the
+          // projection isn't just on the piecewise constants, but onto
+          // the piecewise (bi,tri)linears, and this is going to be a bit
+          // more involved than just computing a single number per shape
+          // function.
+          Assert (this->get_parameters().stokes_velocity_degree==1,
```

Comment:
@tjhei: FYI, this is the place where we trigger the assertion. There is a similar one further down in the file.